components/tabs/src/schema.rs

```diff
@@ -19,13 +22,25 @@ use sql_support::open_database::{
 // fact we are using sql.
 const CREATE_SCHEMA_SQL: &str = "
     CREATE TABLE IF NOT EXISTS tabs (
-        payload TEXT NOT NULL
+        guid      TEXT NOT NULL PRIMARY KEY,
+        payload   TEXT NOT NULL
     );
 ";
 
-pub struct TabsMigrationLogin;
+const CREATE_META_TABLE_SQL: &str = "
+    CREATE TABLE IF NOT EXISTS moz_meta (
+        key    TEXT PRIMARY KEY,
+        value  NOT NULL
+    )
+";
+
+pub(crate) static LAST_SYNC_META_KEY: &str = "last_sync_time";
+pub(crate) static GLOBAL_SYNCID_META_KEY: &str = "global_sync_id";
+pub(crate) static COLLECTION_SYNCID_META_KEY: &str = "tabs_sync_id";
+
+pub struct TabsMigrationLogic;
 
-impl MigrationLogic for TabsMigrationLogin {
+impl MigrationLogic for TabsMigrationLogic {
     const NAME: &'static str = "tabs storage db";
     const END_VERSION: u32 = 1;
```

Comment:
It seems like this should be bumped and we migrate from v1 to v2?

Reply:
We're actually on version 0 right now, since I _think_ this never ran the first time (no DB) the version number didn't get bumped.

Reply:
Does that mean we're never saving the DB to disk right now?
  - If we don't save the DB, then I don't think you need the `v1` migration.  That's only needed for DBs that were created before the `sql_support::open_database` code.
  - If we do save the DB, then I think Mark is right and we need a v1 -> v2 migration.

Reply:
> Does that mean we're never saving the DB to disk right now?

tabs are odd - the db is only created when we need it - which roughly means, when sync has been connected.

Is it possible you are seeing it as version 0 when it doesn't actually exist?

Reply:
I think I didn't have the `PRAGMA version=1` set for the old schema in the test code, which was why I was getting weird results. I think you're both correct and it'll need to be V1 -> V2